controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -72,7 +72,7 @@ const (
 )
 
 // +kubebuilder:rbac:groups=core,resources=events,verbs=create;patch
-// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch
+// +kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete
```

Comment:
Shouldn't this only be on finalizers (as done [this CAPA PR example](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/5722/files))?

Reply:
`ensureCertificatesOwnerRef` indeed sets the owner reference and therefore seems to need `delete` permission as of https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement – @Gacko and me clarified this in chat. Excerpt from the docs:

> This admission controller protects the access to the `metadata.ownerReferences` of an object so that only users with **delete** permission to the object can change it. This admission controller also protects the access to `metadata.ownerReferences[x].blockOwnerDeletion` of an object, so that only users with **update** permission to the finalizers subresource of the referenced _owner_ can change it.

Leaving this comment here for others reviewers as reference.